clang/include/clang/CIR/Dialect/Builder/CIRBaseBuilder.h

```diff
@@ -227,26 +227,24 @@ class CIRBaseBuilderTy : public mlir::OpBuilder {
   //===--------------------------------------------------------------------===//
 
   cir::CallOp createCallOp(mlir::Location loc, mlir::SymbolRefAttr callee,
-                           mlir::Type returnType, mlir::ValueRange operands,
-                           cir::SideEffect sideEffect = cir::SideEffect::All) {
```

Comment:
Have you thought about just having a `ArrayRef<Attribute> attrs = {}`? This would allow the caller to already pass stuff in if available.